src/api/db/structure.sql

```diff
@@ -952,10 +952,12 @@ CREATE TABLE `projects` (
   `kind` enum('standard','maintenance','maintenance_incident','maintenance_release') CHARACTER SET utf8 COLLATE utf8_bin DEFAULT 'standard',
   `url` varchar(255) CHARACTER SET utf8 COLLATE utf8_bin DEFAULT NULL,
   `required_checks` varchar(255) COLLATE utf8mb4_unicode_ci DEFAULT NULL,
+  `staging_workflow_id` int(11) DEFAULT NULL,
```

Comment:
do not modify or bind data to these entries without defining how to store and restore this information from backend.

Reply:
@adrianschroeter I am not sure what you mean, can you please elaborate? :confused:

Reply:
I wonder about somethign else: if you have 2 out of >50.000 projects that actually have a staging workflow, wouldn't it make sense to define the project in the staging workflow and not this way.

Reply:
@coolo  you may misunderstood this, this field is for a 1-M relationship between staging projects and the StagingWorkflow... and as you know staging projects are just projects, so we need to put it here.

The StagingWorkflow table also have a reference to the project (the main one, eg: openSUSE:Factory) to implement the relationship you are talking about.

Please check the models in this PR and let me know if you have any doubts.

Reply:
Our design is:

(Main) Project has one StagingWorkflow.
A StagingWorkflow has many (Staging) Projects.

How you would do this? With a table for Project to Project? And what happen with the model StagingWorkflow? is not just a table is a model. It will also be related to other models, is not just a relationhip.

Reply:
StagingWorkflow:project_id -> Main Project (that part is fine)

StagingProjectsOf:project_id -> Staging Project
StagingProjectsOf:workflow_id -> StagingWorkflow

Project has_one :staging_workflow
Project has_many :staging_projects, through: staging_workflow

I'm sure your design is easier, but I find it also very wasteful on the DB level. And not just that, every Project.find will select one more column by default, which has to be parsed as attribute

Reply:
You are implementing a many to many through... with that structure one StagingProject can be in many Main Projects and that is not true. 

I don't really thing that retrieving one integer field more will slow down too much the queries we already have. I think there are other things to improve in regard to performance, and this one is not that "bad".